library/lvm_gensym.py

```diff
@@ -0,0 +1,165 @@
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+
+lv_name:
+    description: The default generated name for an unspecified logical volume 
+    type: str
+'''
+
+def is_unique(name, used_names):
+    if name not in used_names:
+        return True 
+    
+    return False 
+
+def init_vg_name(host_name, vg_names):
+    def get_os_name():    
+        for line in open('/etc/os-release').readlines():
+            if not line.find('ID='):
+                name = line[3:]
+            
+        name = name.replace('\n', '').replace('"', '')
+        return name 
+
+    # Initialize necessary variables
+    os_name = get_os_name()
+
+    if host_name != None and len(host_name) != 0:
+        vg_default = os_name + '_' + host_name
+    else:
+        vg_default = os_name + '_default' 
+
+    return vg_default
+
+def get_vg_name(host_name, vg_names):
+    name = init_vg_name(host_name, vg_names)
+    counter = 0
+    unique_name = is_unique(name, vg_names)
+
+    while not unique_name:
+        if counter == 0:
+            name = name + '_' + str(counter)
+        else:
+            name = name[:-2] + '_' + str(counter)
+        counter += 1
+        unique_name = is_unique(name, vg_names) 
+
+    vg_names.append(name)
+    return name 
+
+# @param_list = [ fs_type, mount ] both type str 
+def init_lv_name(param_list, lv_names):
+    # Generate name based on supplied input params
+    if 'swap' in param_list[0].lower():
+        lv_default = 'swap'
+    elif param_list[1].startswith('/'):
+        if param_list[1] == '/':
+            lv_default = 'root'
+        else:
+            lv_default = param_list[1][1:].replace('/', '_')
+    else:
+        lv_default = 'mount_default'
+
+    return lv_default 
+    
+def get_lv_name(param_list, lv_names):
+    name = init_lv_name(param_list, lv_names)
+    counter = 0
+    unique_name = is_unique(name, lv_names)
+
+    while not unique_name:
+        if counter == 0:
+            name = name + '_' + str(counter)
+        else:
+            name = name[:-2] + '_' + str(counter)
+        counter += 1
+        unique_name = is_unique(name, lv_names) 
+
+    lv_names.append(name)
+    return name 
+
+def run_module():
+    module_args = dict(
+        mount=dict(type='str', required=True),
+        fs_type=dict(type='str', required=True)
+    )
+
+    result = dict(
+        changed=True,
+        vg_name='',
+        lv_name=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    lvm_facts = facts.ansible_facts(module)['lvm']
+    lv_names = lvm_facts['lvs'].keys()
+    vg_names = lvm_facts['vgs'].keys()
+    host_name = facts.ansible_facts(module)['nodename'].lower().replace('.', '_').replace('-', '_')
+    params = [ module.params['fs_type'], module.params['mount']]
+
+    result['lv_name'] = get_lv_name(params, [element.lower() for element in lv_names])
```

Comment:
It would look cleaner if you just passed in the `lvm_facts` and let `get_lv_name` handle the details.